src/scores/probability/brier_impl.py

```diff
@@ -61,3 +68,155 @@ def brier_score(
+        obs: Real valued observations variables in xarray.These values are converted to binary
+            values in this function.
+        ensemble_member_dim: The dimension name of the ensemble member.
+        event_thresholds: The threshold(s) that define what is considered an event. If
+            multiple thresholds are provided, they should be monotonically increasing with no NaNs
+        reduce_dims: Optionally specify which dimensions to reduce when
+            calculating the Brier score. All other dimensions will be preserved.
+        preserve_dims: Optionally specify which dimensions to preserve when
+            calculating the score. All other dimensions will be reduced. As a
+            special case, 'all' will allow all dimensions to be preserved. In
+            this case, the result will be in the same shape/dimensionality
+            as the forecast, and the errors will be the absolute error at each
+            point (i.e. single-value comparison against observed), and the
+            forecast and observed dimensions must match precisely.
+        weights: Optionally provide an array for weighted averaging (e.g. by area, by latitude)
+            when calculating the mean score across specified dimensions via ``reduce_dims`` or ``preserve_dims``.
+        fair_correction: Whether or not to apply the fair Brier score correction. Default is True.
+        event_threshold_mode: "The mode to use to define an event relative to a supplied threshold.
+            Default is ``operator.ge``, which means that an event occurs (is forecast)
+            if the observation (forecast) is greater than or equal to the threshold." Passing
+            in ``operator.lt`` will give the same results. Alternatively, you can provide
+            ``operator.gt`` which means that an event occurs (is forecast) if the observation (forecast)
+            is strictly greater than the threshold. Using``operator.le`` will give the same results as ``operator.gt``.
+        threshold_dim: The name of the threshold dimension in the output. Default is 'threshold'.
+           It must not exist as a dimension in the forecast or observation data.
+
+
+    Returns:
+        The Brier score for the ensemble forecast.
+
+    Raises:
+        ValueError: if values in ``event_thresholds`` are not monotonically increasing.
+        ValueError: if ``fcst`` contains the dimension provided in ``threshold_dim``.
+        ValueError: if ``obs`` contains the dimension provided in ``threshold_dim``.
+        ValueError: if ``weights`` contains the provided in ``threshold_dim``.
+        ValueError: if ``ensemble_member_dim`` is not in ``fcst`` dimensions.
+
+    References:
+        - Ferro, C. A. T. (2013). Fair scores for ensemble forecasts. Quarterly
+            Journal of the Royal Meteorological Society, 140(683), 1917–1923.
+            https://doi.org/10.1002/qj.2270
+
+    See Also:
+        - :py:func:`scores.probability.brier_score`
+
+    Examples:
```

Comment:
I'm wondering if the example would be slight clearer if the forecast and obs values were sampled from something other than the unit interval.

The reason is that the Brier score usually applies for probability forecasts (in [0,1]) but in this example, you might want to illustrated that the forecasts aren't probability forecasts.

Could be easy to change by multiplying fcst,obs by 10 and adding 20 (or something like that)